packages/project/lib/specifications/types/Component.js

```diff
@@ -0,0 +1,303 @@
+import fsPath from "node:path";
+import posixPath from "node:path/posix";
+import ComponentProject from "../ComponentProject.js";
+import {createReader} from "@ui5/fs/resourceFactory";
+
+/**
+ * Component
+ *
+ * @public
+ * @class
+ * @alias @ui5/project/specifications/types/Component
+ * @extends @ui5/project/specifications/ComponentProject
+ * @hideconstructor
+ */
+class Component extends ComponentProject {
+	constructor(parameters) {
+		super(parameters);
+
+		this._pManifests = Object.create(null);
+
+		this._srcPath = "src";
+		this._testPath = "test";
+		this._testPathExists = false;
+
+		this._propertiesFilesSourceEncoding = "UTF-8";
+	}
+
+	/* === Attributes === */
+
+	/**
+	* Get the cachebuster signature type configuration of the project
+	*
+	* @returns {string} <code>time</code> or <code>hash</code>
+	*/
+	getCachebusterSignatureType() {
+		return this._config.builder && this._config.builder.cachebuster &&
+			this._config.builder.cachebuster.signatureType || "time";
+	}
+
+	/**
+	 * Get the path of the project's source directory. This might not be POSIX-style on some platforms.
+	 *
+	 * @public
+	 * @returns {string} Absolute path to the source directory of the project
+	 */
+	getSourcePath() {
+		return fsPath.join(this.getRootPath(), this._srcPath);
+	}
+
+	getSourcePaths() {
+		const paths = [this.getSourcePath()];
+		if (this._testPathExists) {
+			paths.push(fsPath.join(this.getRootPath(), this._testPath));
+		}
+		return paths;
+	}
+
+	getVirtualPath(sourceFilePath) {
+		const sourcePath = this.getSourcePath();
+		if (sourceFilePath.startsWith(sourcePath)) {
+			const relSourceFilePath = fsPath.relative(sourcePath, sourceFilePath);
+			let virBasePath = "/resources/";
+			if (!this._isSourceNamespaced) {
+				virBasePath += `${this._namespace}/`;
+			}
+			return posixPath.join(virBasePath, relSourceFilePath);
+		}
+
+		const testPath = fsPath.join(this.getRootPath(), this._testPath);
+		if (sourceFilePath.startsWith(testPath)) {
+			const relSourceFilePath = fsPath.relative(testPath, sourceFilePath);
+			let virBasePath = "/test-resources/";
+			if (!this._isSourceNamespaced) {
+				virBasePath += `${this._namespace}/`;
+			}
+			return posixPath.join(virBasePath, relSourceFilePath);
+		}
+
+		throw new Error(
+			`Unable to convert source path ${sourceFilePath} to virtual path for project ${this.getName()}`);
+	}
+
+	/* === Resource Access === */
+	/**
+	* Get a resource reader for the sources of the project (excluding any test resources)
+	*
+	* @param {string[]} excludes List of glob patterns to exclude
+	* @returns {@ui5/fs/ReaderCollection} Reader collection
+	*/
+	_getSourceReader(excludes) {
+		return createReader({
```

Comment:
Why always creating a new reader?
Is it possible that this reader can be cached?

Reply:
Indeed caching could be done for readers here and elsewhere. Note that this code is almost identical to the other project types, so I would see it independent fro this PR.

Maybe we can investigate this topic in conjunction with working on [RFC 0017 Incremental Build](https://github.com/UI5/cli/blob/rfc-incremental-build/rfcs/0017-incremental-build.md)